spec/services/users/create_spec.rb

```diff
@@ -2,5 +2,64 @@
 require 'rails_helper'
 
 RSpec.describe Users::Create, type: :service do
-  # TODO: write your tests here. You can inspire by this code: https://github.com/mateuszbialowas/brug-service_objects/pull/1/files
+  subject(:service) { described_class.new(user_params:, shop_params:).call }
+  let(:industry) { create(:industry) }
+  let(:industry_id) { industry.id }
+  let(:user_params) { { name: 'John', email: 'john2n@gmail.com' } }
+  let(:shop_params) { { industry_id:, name: 'John Shop' } }
+
+  context 'when shop_params are invalid' do
+    let(:industry_id) { 0 }
+
+    it 'returns failure' do
+      expect { service }.to not_change { User.count }.and not_change { Shop.count }
+      expect(service.failure).to eq(industry_id: ['Industry not found'])
+    end
+  end
+
+  context 'when user params are invalid' do
+    let(:user_params) { { name: nil, email: nil } }
+
+    it 'returns failure' do
+      expect { service }.to not_change { User.count }.and not_change { Shop.count }
+      expect(service.failure).to eq({ name: ['must be filled'], email: ['must be filled'] })
+    end
+  end
+
+  context 'when error creating shop' do
+    before do
+      allow_any_instance_of(Shop).to receive(:save).and_return(false)
+      allow_any_instance_of(Shop).to receive(:errors).and_return(double(full_messages: ['Shop error']))
+    end
+
+    it 'returns failure' do
+      expect { service }.to not_change { User.count }.and not_change { Shop.count }
+      expect(service.failure).to eq('Shop error')
+    end
+  end
+
+  context 'when error creating user' do
+    let(:user_double) { instance_double(User, save: false, errors: double(full_messages: ['User error'])) }
```

Comment:
w przypadku mockowania errorow powinnismy dawac expect(blabla).to_receive(blabla).and_raise(ValidationError/CostamError) imho. Ale mozna to tez przedyskutowac

Reply:
zgodze sie z tym ze tu da sie inaczej. 
Nie znam tego gema validacji i jak dziala ale w kodzie validatora jest text 'is not unique' ktorego bym sie spodziewal zamiast User Error.
Bo troche dziwnie tubowac string a potem robic asserta ze sie go otrzymalo ale tez zalezy jakie flow idzie

Reply:
ten error ma sie pojawić jak user.save zwróci false. To nie dotyczy walidacji. Nie umiem powiedzieć kiedy ten błąd może zaistnieć, ale to jest tak jakby testowanie na przyszłość. Jeżeli kiedyś save zwróci false, no to user nie będzie miał 500.

Reply:
![image](https://user-images.githubusercontent.com/58574619/214012715-1d0aa684-6dd5-465a-90cb-914c49c88f2b.png)
tu testuję walidację

Reply:
tak naprawdę to na chwilę obecną ten test nic nie sprawdza, bo w obecnym kodzie nie ma możliwości, żeby user.save zwróciło false. To było napisane, zeby pokryć 100% kodu i co jeżeli save zwróci false, pomimo, że no nie zwróci.